scene/gui/code_edit.cpp

```diff
@@ -1756,6 +1806,74 @@ TypedArray<int> CodeEdit::get_folded_lines() const {
 	return folded_lines;
 }
 
+/* Code region */
+void CodeEdit::create_code_region() {
+	// Abort if there is no selected text
+	if (!has_selection()) {
+		return;
+	}
+	// Check that region tag find a comment delimiter and is valid
+	if (code_region_start.is_empty()) {
+		WARN_PRINT_ONCE("Cannot create code region without any one line comment delimiters");
+		return;
+	}
+	begin_complex_operation();
+	// Adding start and end region tags
+	int first_region_start = -1;
+	for (int caret_idx : get_caret_index_edit_order()) {
+		if (!has_selection(caret_idx)) {
+			continue;
+		}
+		int from_line = get_selection_from_line(caret_idx);
+		if (first_region_start == -1 || from_line < first_region_start) {
+			first_region_start = from_line;
+		}
+		int to_line = get_selection_to_line(caret_idx);
+		set_line(to_line, get_line(to_line) + "\n" + code_region_end);
+		insert_line_at(from_line, code_region_start + " " + RTR("New Code Region"));
+		fold_line(from_line);
+	}
+
+	// Select name of the first region to allow quick edit
+	remove_secondary_carets();
+	set_caret_line(first_region_start);
+	int tag_length = code_region_start.length() + RTR("New Code Region").length() + 1;
+	set_caret_column(tag_length);
+	select(first_region_start, code_region_start.length() + 1, first_region_start, tag_length);
+
+	end_complex_operation();
+	queue_redraw();
+}
+
+String CodeEdit::get_code_region_start_tag() const {
+	return code_region_start_tag;
+}
+
+String CodeEdit::get_code_region_end_tag() const {
+	return code_region_end_tag;
+}
+
+void CodeEdit::set_code_region_tags(const String &p_start, const String &p_end) {
+	ERR_FAIL_COND_MSG(p_start == p_end, "Starting and ending region tags cannot be identical.");
+	code_region_start_tag = p_start;
+	code_region_end_tag = p_end;
+	_update_code_region_tags();
+}
+
+bool CodeEdit::is_line_code_region_start(int p_line) const {
```

Comment:
Should check `ERR_FAIL_INDEX_V(p_line, get_line_count(), false);` same in `is_line_code_region_end`